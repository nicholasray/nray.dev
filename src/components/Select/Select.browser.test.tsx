import { render } from "vitest-browser-react";
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from "./Select";

function ThemeSelect() {
  return (
    <Select>
      <SelectTrigger showIcon={false} className="w-[180px]">
        <SelectValue placeholder="Theme" />
      </SelectTrigger>
      <SelectContent>
        <SelectItem value="light">Light</SelectItem>
        <SelectItem value="dark">Dark</SelectItem>
        <SelectItem value="system">System</SelectItem>
      </SelectContent>
    </Select>
  );
}

it("renders correctly", async () => {
  const screen = render(<ThemeSelect />);

  await expect.element(screen.getByRole("combobox")).toBeInTheDocument();

  expect(screen.container).toMatchSnapshot();
});
