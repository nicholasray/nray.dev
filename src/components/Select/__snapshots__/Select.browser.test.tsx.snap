// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<div>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r0:"
    aria-expanded="false"
    class="flex h-9 items-center justify-between whitespace-nowrap rounded-md border border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 w-[180px]"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      Theme
    </span>
  </button>
</div>
`;
